src/main/java/it/okkam/validation/FiscalCodeValidator.java

```diff
@@ -321,51 +321,39 @@ private static String calculateControlChar(String result) {
     for (int i = 1; i <= 13; i += 2) {
       Integer valueToSum = evenSumValues.get(String.valueOf(result.charAt(i)));
       if (valueToSum == null) {
-        throw new IllegalArgumentException(
-            "Control character " + result.charAt(i) + " unsupported value");
+        throw new IllegalArgumentException(CONTROL + result.charAt(i) + UNSUPPORTED);
       }
       evenSum += valueToSum;
     }
     int oddSum = 0;
     for (int i = 0; i <= 14; i += 2) {
       Integer valueToSum = oddSumValues.get(String.valueOf(result.charAt(i)));
       if (valueToSum == null) {
-        throw new IllegalArgumentException(
-            "Control character " + result.charAt(i) + " unsupported value");
+        throw new IllegalArgumentException(CONTROL + result.charAt(i) + UNSUPPORTED);
       }
       oddSum += valueToSum;
     }
     int interoControllo = (evenSum + oddSum) % 26;
     String controlChar = controlCharValues.get(interoControllo);
     if (controlChar == null) {
-      throw new IllegalArgumentException(
-          "Control character " + interoControllo + " unsupported value");
+      throw new IllegalArgumentException(CONTROL + interoControllo + UNSUPPORTED);
     }
     return controlChar;
   }
 
   /**
    * Initialize the FiscalCodeConf.
    * 
-   * @param codiciIstatStr
-   *          the string content of the TSV containing CODICE-ISTAT => TOWN mappings
-   * @param maxComuneNameLength
-   *          the max length of the name of a comune (0 to disable generation of truncated version
-   *          of the name)
-   * @param maleValue
-   *          the String denoting a male (e.g. "M", "MALE", etc..)
-   * @param yearStart
-   *          year start index
-   * @param yearEnd
-   *          year end index
-   * @param monthStart
-   *          month start index
-   * @param monthEnd
-   *          month end index
-   * @param dayStart
-   *          day start index
-   * @param dayEnd
-   *          day end index
+   * @param codiciIstatStr the string content of the TSV containing CODICE-ISTAT => TOWN mappings
+   * @param maxComuneNameLength the max length of the name of a comune (0 to disable generation of
+   *        truncated version of the name)
+   * @param maleValue the String denoting a male (e.g. "M", "MALE", etc..)
+   * @param yearStart year start index
+   * @param yearEnd year end index
+   * @param monthStart month start index
+   * @param monthEnd month end index
+   * @param dayStart day start index
+   * @param dayEnd day end index
    * @return the corresponding FiscalCodeConf bean
    */
   public static FiscalCodeConf getFiscalCodeConf(String codiciIstatStr, int maxComuneNameLength,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS00107)